src/iexamine.cpp

```diff
@@ -2280,6 +2280,164 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
     none( p, examp );
 }
 
+item_location maple_tree_sap_container() {
+    const item maple_sap = item( "maple_sap", 0 );
+    return g->inv_map_splice( [&]( const item &it ) { 
+        return it.has_valid_capacity_for_liquid( maple_sap, true ) == item::LIQUID_FILL_ERROR::L_ERR_NONE;
+    }, _( "Which container:" ), PICKUP_RANGE );
+}
+
+void iexamine::tree_maple(player &p, const tripoint &examp)
+{
+    if( !p.has_items_with_quality( "DRILL", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to drill the crust to tap this maple tree." ) );
+        return;
+    }
+
+    if( !p.has_items_with_quality( "HAMMER", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to hammer the spile into the crust to tap this maple tree." ) );
+        return;
+    }
+
+    const inventory &crafting_inv = p.crafting_inventory();
+
+    if( !crafting_inv.has_amount( "tree_spile", 1 ) ) {
+        std::string spile_name = item( "tree_spile", 0 ).tname( 1 );
+        add_msg( m_info, _( "You need a %s to tap this maple tree." ), spile_name.c_str() );
```

Comment:
To get the name of an item, you can use `item::nname`. It's a static function, it requires only the id (quantity is optional and defaults to 1, so it's not needed).